src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepWorkflowTest.java

```diff
@@ -0,0 +1,112 @@
+                        new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(getPrivateKey()), "cloudbees", "test");
+                SystemCredentialsProvider.getInstance().getCredentials().add(key);
+                SystemCredentialsProvider.getInstance().save();
+
+                WorkflowJob job = story.j.jenkins.createProject(WorkflowJob.class, "sshAgentAvailable");
+                job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  sshagent ([ignoreMissing: false, credentials: ['" + CREDENTIAL_ID + "']]) {\n"
+                        + "    sh 'ssh -o StrictHostKeyChecking=no -p " + getAssignedPort() + " -v -l cloudbees " + SSH_SERVER_HOST + "'\n"
+                        + "  }\n"
+                        + "}\n", true)
+                );
+                story.j.assertBuildStatusSuccess(job.scheduleBuild2(0));
+
+                stopMockSSHServer();
+            }
+        });
+    }
+
+    @Test
+    public void sshAgentAvailableAfterRestart() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                startMockSSHServer();
+
+                List<String> credentialIds = new ArrayList<String>();
+                credentialIds.add(CREDENTIAL_ID);
+
+                SSHUserPrivateKey key = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, credentialIds.get(0), "cloudbees",
+                        new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(getPrivateKey()), "cloudbees", "test");
+                SystemCredentialsProvider.getInstance().getCredentials().add(key);
+                SystemCredentialsProvider.getInstance().save();
+
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "sshAgentAvailableAfterRestart");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  sshagent ([ignoreMissing: false, credentials: ['" + CREDENTIAL_ID + "']]) {\n"
+                        + "    sh 'ssh -o StrictHostKeyChecking=no -p " + getAssignedPort() + " -v -l cloudbees " + SSH_SERVER_HOST + "'\n"
+                        + "    semaphore 'sshAgentAvailableAfterRestart'\n"
+                        + "    sh 'ssh -o StrictHostKeyChecking=no -p " + getAssignedPort() + " -v -l cloudbees " + SSH_SERVER_HOST + "'\n"
+                        + "  }\n"
+                        + "}\n", true));
+                // get the build going
+                WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+                CpsFlowExecution e = (CpsFlowExecution) b.getExecutionPromise().get();
+
+                // wait until the executor gets assigned and the execution pauses
+                SemaphoreStep.waitForStart("sshAgentAvailableAfterRestart/1", b);
+                assertTrue(JenkinsRule.getLog(b), b.isBuilding());
```

Comment:
I think at this point you want to forcibly stop any running server. Otherwise the test would probably pass even with `onResume` deleted. Right?